better_exceptions/__init__.py

```diff
@@ -279,7 +279,12 @@ def format_traceback_frame(tb):
         if not PY3 and isinstance(val, str):
             # In Python2 the Non-ASCII value will be the escaped string,
             # use string-escape to decode the string to show the text in human way.
-            val = _unicode(val.decode("string-escape"))
+            # If string is very long, the val will only contians part of the string,
+            # that will cause decode error, so we just decode the complete string
+            # which ends with quote.
+            if val.endswith(val[0]):
```

Comment:
The complete string will starts and ends with quote, the quote could be single(`'`) or double(`"`). But the first character of the string will always is the quote,  so with this check we don't need care about what kind the quote is.

PS. English is not my native language; please excuse typing errors.